includes/Core/Authentication/Authentication.php

```diff
@@ -406,6 +406,26 @@ function() {
 				$this->maybe_refresh_token_for_screen( $this->context->input()->filter( INPUT_POST, 'screen_id' ) );
 			}
 		);
+
+		// Regularly synchronize Google profile data.
+		add_action(
+			'googlesitekit_reauthorize_user',
+			function() {
+				if ( ! $this->profile->has() ) {
+					return;
+				}
+
+				$profile_data = $this->profile->get();
+
+				if ( ! isset( $profile_data['last_updated'] ) ) {
+					return;
+				}
```

Comment:
If the `last_updated` value is not set, we should refresh profile data as well. Otherwise, profiles that don't have that property will never be refreshed.

Reply:
Excellent catch. I've updated this to refresh the profile data if `last_updated` doesn't exist. Thank you, @eugene-manuilov!